app/Console/Commands/Indexer/IncrementalIndexer.php

```diff
@@ -30,6 +30,23 @@ protected function promptForMissingArgumentsUsing(): array
         ];
     }
 
+    private function sleep(int $milliseconds): void
+    {
+        $interval = 100; // check every 100 ms
+        $elapsed = 0;
+
+        while ($elapsed < $milliseconds)
+        {
+            if ($this->cancelled)
+            {
+                return;
+            }
+
+            usleep($interval * 1000);
+            $elapsed += $interval;
+        }
+    }
+
```

Comment:
@pushrbx Is this hack specific to docker containers?

Reply:
Also, would it make sense to move this to helpers?

Reply:
I just wanted to have a sleep function which can be cancelled. Anytime you cancel the process, it should gracefully exit.

Reply:
I'll move this to helpers.